tests/test_datacollection.py

```diff
@@ -0,0 +1,197 @@
+from eitprocessing.datahandling.continuousdata import ContinuousData
+from eitprocessing.datahandling.datacollection import DataCollection
+from eitprocessing.datahandling.eitdata import EITData
+from eitprocessing.datahandling.intervaldata import IntervalData
+from eitprocessing.datahandling.sparsedata import SparseData
+
+
+@pytest.fixture
+def create_data_object() -> Callable[[str, list | None], ContinuousData]:
+    def internal(
+        label: str,
+        derived_from: list | None = None,
+        time: np.ndarray | None = None,
+        values: np.ndarray | None = None,
+    ) -> ContinuousData:
+        return ContinuousData(
+            label=label,
+            name="",
+            unit="",
+            category="physical measurements",
+            time=time if isinstance(time, np.ndarray) else np.array([]),
+            sample_frequency=0,
+            values=values if isinstance(values, np.ndarray) else np.array([]),
+            derived_from=derived_from if isinstance(derived_from, list) else [],
+        )
+
+    return internal
+
+
+def test_init():
+    _ = DataCollection(ContinuousData)
+    _ = DataCollection(IntervalData)
+    _ = DataCollection(EITData)
+    _ = DataCollection(SparseData)
+
+    with pytest.raises(TypeError):
+        _ = DataCollection(str)
+
+    with pytest.raises(TypeError):
+        _ = DataCollection(DataContainer)
+
+
+def test_check_item(create_data_object: Callable[[str], ContinuousData]):
+    data_object = create_data_object(label="test label")
+    data_object_b = create_data_object(label="test label")
+
+    dc = DataCollection(ContinuousData)
+
+    dc._check_item(data_object, key="test label")
+    dc._check_item(data_object)
+
+    with pytest.raises(KeyError):
+        # key does not match label
+        dc._check_item(data_object, key="not test label")
+
+    with pytest.raises(TypeError):
+        # object type does not match
+        dc._check_item("string")
+
+    # do not use dc.__setitem__() (dc[...] = ...) which uses dc._check_item()
+    UserDict.__setitem__(dc, "test label", data_object)
+
+    assert "test label" in dc
+
+    with pytest.raises(KeyError):
+        # do not allow overwriting with the same label
+        dc._check_item(data_object_b)
+
+    dc._check_item(data_object_b, overwrite=True)
+
+
+def test_check_add(create_data_object: Callable[[str], ContinuousData]):
+    data_object_1 = create_data_object("label 1")
+    data_object_1_b = create_data_object("label 1")
+    data_object_2 = create_data_object("label 2")
+    data_object_3 = create_data_object("label 3")
+
+    dc = DataCollection(ContinuousData)
+    assert len(dc) == 0
+
+    dc.add(data_object_1)
+    assert "label 1" in dc
+    assert len(dc) == 1
+
+    dc.add(data_object_2, data_object_3)
+    assert "label 2" in dc
+    assert "label 3" in dc
+    assert len(dc) == 3
+
+    with pytest.raises(KeyError):
+        dc.add(data_object_1_b)
+
+    dc.add(data_object_1_b, overwrite=True)
```

Comment:
Wouldn't it make sense to assert whether this overwrite acts as expected (check length of dc, check whether dc['label 1'] is now data_object_1_b)? To me it seems weird to do this overwrite here but not add any assertions but maybe I am missing something.

Reply:
The same for the last line of test_check_item

Reply:
`test_check_item` does not change `dc`, so there is nothing to assert there.